internal/driver/node.go

```diff
@@ -262,48 +251,32 @@ func (s *nodeServerNoLocked) nodePublishFilesystemVolume(req *csi.NodePublishVol
 	return nil
 }
 
-func (s *nodeServerNoLocked) createDeviceIfNeeded(device string, lv *proto.LogicalVolume) error {
-	var stat unix.Stat_t
-	err := filesystem.Stat(device, &stat)
-	switch err {
-	case nil:
-		// a block device already exists, check its attributes
-		if stat.Rdev == unix.Mkdev(lv.DevMajor, lv.DevMinor) && stat.Uid == uint32(os.Getuid()) && stat.Mode == deviceMode {
-			return nil
-		}
-		err := os.Remove(device)
-		if err != nil {
-			return status.Errorf(codes.Internal, "failed to remove device file %s: error=%v", device, err)
-		}
-		fallthrough
-	case unix.ENOENT:
-		err = os.MkdirAll(path.Dir(device), 0755)
-		if err != nil {
-			return status.Errorf(codes.Internal, "mkdir failed: target=%s, error=%v", path.Dir(device), err)
-		}
-
-		devno := unix.Mkdev(lv.DevMajor, lv.DevMinor)
-		if err := filesystem.Mknod(device, deviceMode, int(devno)); err != nil {
-			return status.Errorf(codes.Internal, "mknod failed for %s. major=%d, minor=%d, error=%v",
-				device, lv.DevMajor, lv.DevMinor, err)
-		}
-	default:
-		return status.Errorf(codes.Internal, "failed to stat %s: error=%v", device, err)
-	}
-	return nil
-}
-
 func (s *nodeServerNoLocked) nodePublishBlockVolume(req *csi.NodePublishVolumeRequest, lv *proto.LogicalVolume) error {
 	// Find lv and create a block device with it
-	targetPath := req.GetTargetPath()
-	err := s.createDeviceIfNeeded(targetPath, lv)
+	mountOption := req.GetVolumeCapability().GetMount()
```

Comment:
I agree this code was a bit misleading, changed it so that it only uses the readonly option from the mount options and no longer needs the VolumeCapabilityMount